type.md

```diff
@@ -0,0 +1,175 @@
+---
+title: Type System
+---
+
+Common Lisp has a complete and flexible type system and corresponding tools to
+inspect, check and manipulate types.
+
+## Values Have Types, Not Variables
+
+Being different from some languages such as C/C++, variables in Lisp are just
+*placeholders* for values. When you [`SETF`][setf] something, the value is only
+*bound* to the variable. You can bind another value to the same variable later,
+as you wish.
+
+This implies a fact that in Common Lisp **values have types**, while variables
+do not. This might be surprising at first if you come from a C/C++ background.
+
+For example:
+
+~~~lisp
+* (defvar *var* 1234)
+*VAR*
+
+* (type-of *var*)
+(INTEGER 0 4611686018427387903)
+~~~
+
+Function [`TYPE-OF`][type-of] returns the type of given variable. The returned
+result is a [type specifier][type-specifier]. The first element is the type and
+the remaining part is extra information of that type. In this case it is lower
+bound and upper bound. You can safely ignore it for now. Also remember that
+numbers in Lisp have no limit!
+
+Now let's try to [`SETF`][setf] the variable:
+
+~~~lisp
+* (setf *var* "hello")
+"hello"
+
+* (type-of *var*)
+(SIMPLE-ARRAY CHARACTER (5))
+~~~
+
+You see, the type of the same variable is changed to
+[`SIMPLE-ARRAY`][simple-array], with contents of type [`CHARACTER`][character]
+and length 5.
+
+## Type Hierarchy
+
+Inheritance relationship of Lisp types consists a type tree. For example:
+
+~~~lisp
+* (describe 'number)
+COMMON-LISP:NUMBER
+  [symbol]
+
+NUMBER names the built-in-class #<BUILT-IN-CLASS COMMON-LISP:NUMBER>:
+  Class precedence-list: NUMBER, T
+  Direct superclasses: T
+  Direct subclasses: COMPLEX, REAL
+  No direct slots.
+~~~
+
+Function [`DESCRIBE`][describe] shows that type [`NUMBER`][number] inherits type
+`T` and is directly inherited by types [`COMPLEX`][complex] and [`REAL`][real].
+
+Similarly, the precedence list of type [`INTEGER`][integer] is
+[`INTEGER`][integer] <- [`RATIONAL`][rational] <- [`REAL`][real] <-
+[`NUMBER`][number] <- `T`. Type `T` is the root of **all** types.
+
+It might be confusing that [`NUMBER`][number] is a built-in-class. This is
+because Lisp types are implemented as CLOS classes. For example:
+
+~~~lisp
+* (type-of 1234)
+(INTEGER 0 4611686018427387903)
+
+* (class-of 1234)
+#<BUILT-IN-CLASS COMMON-LISP:FIXNUM>
+~~~
+
+Function [`CLASS-OF`][class-of] gives more specific result.
+
+## Work with Types
+
+Function [`TYPEP`][typep] can be used to check if given argument is of given
+type.
+
+~~~lisp
+* (typep 1234 'integer)
+T
+~~~
+
+Function [`SUBTYPEP`][subtypep] can be used to inspect if a type is sub-type of
```

Comment:
THE function… 

or just

`subtypep` can be used… is THE sub-type

Reply:
ACK.